### ISSUE TITLE
filter particle by size slider crashes

### ISSUE THREAD

**Issue by chrishavlin** [yt-project/yt_idv#152]
Running either particle plot example, trying to use the "Filter Particle Max Size" slider causes a crash

https://github.com/user-attachments/assets/953ae280-8a37-4201-a1e9-4fa2b1a82010

```python-traceback
Traceback (most recent call last):
  File "/home/chavlin/src/yt_general/website_explore/data_structure_figures/discrete_particles_idv.py", line 19, in <module>
    rc.run()
  File "/home/chavlin/src/yt_general/yt_idv/yt_idv/rendering_contexts/pyglet_context.py", line 158, in run
    pyglet.app.run()
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/app/__init__.py", line 76, in run
    event_loop.run(interval)
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/app/base.py", line 151, in run
    timeout = self.idle()
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/app/base.py", line 212, in idle
    self.clock.call_scheduled_functions(dt)
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/clock.py", line 227, in call_scheduled_functions
    item.func(now - item.last_ts, *item.args, **item.kwargs)
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/app/base.py", line 118, in _redraw_windows
    window.dispatch_event('on_draw')
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/window/__init__.py", line 668, in dispatch_event
    super().dispatch_event(*args)
  File "/home/chavlin/.pyenv/versions/yt_idv/lib/python3.9/site-packages/pyglet/event.py", line 387, in dispatch_event
    if getattr(self, event_type)(*args):
  File "/home/chavlin/src/yt_general/yt_idv/yt_idv/rendering_contexts/pyglet_context.py", line 80, in on_draw
    self.gui.render(self.scene)
  File "/home/chavlin/src/yt_general/yt_idv/yt_idv/simple_gui.py", line 63, in render
    if imgui.tree_node(f"element {i + 1}: {element.display_name}"):
  File "imgui/core.pyx", line 5130, in imgui.core.tree_node
imgui.core.ImGuiError: ImGui assertion error (mouse_button >= 0 && mouse_button < ImGuiMouseButton_COUNT) at imgui-cpp/imgui_widgets.cpp:627

```

I actually think this is an `imgui` bug -- downgrading from imgui 2.0.0 to 1.4.1 fixes the error for me. but reporting here til I can figure out a more minimally reproducible example.

**Comment by matthewturk** [comment]
Well isn't that strange!  I wonder if this is just related to the specific UI element?

**Comment by chrishavlin** [comment]
ya, i think it is just a bug with the UI element, most likely not really a yt_idv bug.

**Comment by chrishavlin** [comment]
Fixed by #153
